tests/test_linkextractors.py

```diff
@@ -851,3 +852,36 @@ def test_skip_bad_links(self):
                 ),
             ],
         )
+
+    def test_link_allowed_is_false_with_empty_url(self):
+        mock_link = Mock()
```

Comment:
Do you need to use `Mock` in these tests? It feels like `Link` could work, and `Mock` is usually left for cases where it is needed.